great_expectations/datasource/datasource.py

```diff
@@ -115,48 +138,55 @@ def list_generators(self):
         return [{"name": key, "type": value["type"]} for key, value in self._datasource_config["generators"].items()]
 
     def get_data_asset(self, data_asset_name, batch_kwargs=None, **kwargs):
```

Comment:
The method is called get_data_asset, but it returns a batch. Are we ok with the naming discrepancy?

Reply:
Good catch. Changed.